pandas/core/groupby/groupby.py

```diff
@@ -1211,10 +1211,10 @@ def _aggregate_with_numba(self, data, func, *args, engine_kwargs=None, **kwargs)
         if cache_key not in NUMBA_FUNC_CACHE:
             NUMBA_FUNC_CACHE[cache_key] = numba_agg_func
 
-        if self.grouper.nkeys > 1:
-            index = MultiIndex.from_tuples(group_keys, names=self.grouper.names)
-        else:
+        if self.grouper.nkeys <= 1:
```

Comment:
Mypy binds the type early. I changed the order: Index first, then MultiIndex.

Reply:
not sure I understand what the consequence of this is, could you elaborate please?

Reply:
Previously, mypy assigned `index: MultiIndex` which then fails when assigning `Index` to `index`. I think that `MultiIndex.from_tuples` was un-typed because of a decorator, so that wasn't detected earlier.

Reply:
cool, thanks for explaining!